File: Java/Binary_Search.java
public class Binary_Search {
	int BinarySearch(int arr[], int l, int r, int x){
		if (r>0){
			int mid = l + (r-l)/2;
			
			if (arr[mid]==x){
				return (mid);
			}
			else if (arr[mid]>x){
				return BinarySearch(arr, l, mid-1, x);
			}
			else{
				return BinarySearch(arr, mid+1, r, x);
			}
		}
		return (-1);
		
	}
}

File: C/binary_search.c
#include <stdio.h>

/*Recursive function to run the binary search*/
int binary_search(unsigned int arr[], int l, int r, unsigned int data){
    if (r>=l){
        int mid = l +(r-l)/2; /*Choosing the middle element*/

        if (arr[mid]==data){
            /*If the current middle element is the element we need*/
            return mid;
        }
        if (arr[mid]>data){
            /*If our element is on the left hand side of the middle element*/
            return binary_search(arr, l, mid-1, data);
        }
        /*If our element is on the right hand side of the middle element*/
        return binary_search(arr,mid+1,r,data);
    }
    /*If r<l, then the element is not in the array*/
    return -1;
}
